ee/tuf/library_lookup.go

```diff
@@ -52,21 +53,57 @@ func CheckOutLatestWithoutConfig(binary autoupdatableBinary, logger log.Logger)
 }
 
 func UsingNewAutoupdater() bool {
-	cfg, err := getAutoupdateConfig()
+	cfg, err := getAutoupdateConfig(os.Args[1:])
 	if err != nil {
 		return false
 	}
 
 	return ChannelUsesNewAutoupdater(cfg.channel)
 }
 
-// getAutoupdateConfig reads launcher's config file to determine the configuration values
-// needed to work with the autoupdate library.
-func getAutoupdateConfig() (*autoupdateConfig, error) {
-	configFilePath := launcher.ConfigFilePath(os.Args[1:])
-	if configFilePath == "" {
-		return nil, errors.New("could not get config file path")
+// getAutoupdateConfig pulls the configuration values necessary to work with the autoupdate library
+// from either the given args or from the config file.
+func getAutoupdateConfig(args []string) (*autoupdateConfig, error) {
+	// Create a flagset with options that are relevant to autoupdate only.
+	// Ensure that we won't fail out when we see other command-line options.
+	pflagSet := pflag.NewFlagSet("autoupdate options", pflag.ContinueOnError)
+	pflagSet.ParseErrorsWhitelist = pflag.ParseErrorsWhitelist{UnknownFlags: true}
+
+	// Extract the config flag plus the autoupdate flags
+	var flConfigFilePath, flRootDirectory, flUpdateDirectory, flUpdateChannel, flLocalDevelopmentPath string
+	pflagSet.StringVar(&flConfigFilePath, "config", "", "")
+	pflagSet.StringVar(&flRootDirectory, "root_directory", "", "")
+	pflagSet.StringVar(&flUpdateDirectory, "update_directory", "", "")
+	pflagSet.StringVar(&flUpdateChannel, "update_channel", "", "")
+	pflagSet.StringVar(&flLocalDevelopmentPath, "localdev_path", "", "")
+
+	if err := pflagSet.Parse(args); err != nil {
+		return nil, fmt.Errorf("parsing command-line flags: %w", err)
+	}
+
+	// If the config file wasn't set AND the other flags weren't set, fall back
+	// to looking in the default config flag file location
+	if flConfigFilePath == "" && flRootDirectory == "" && flUpdateChannel == "" {
```

Comment:
should we also be checking for `flLocalDevelopmentPath` here?

Reply:
or if it's set, seems like we need a way to add that to what's returned from `getAutoupdateConfigFromFile`

Reply:
We have to have either `--config`, or `--root_directory` and `--update_channel` at the minimum, so that's what I'm checking for here -- if none of them are set, then we need to check in the well-known config file location for a config instead. `--localdev_path` and `--update_directory` are both optional so I didn't check for them here.

I think I'm not super concerned with the edge case where someone runs launcher in a weird way where they pass in `--config` with some of the values set, but specifies `--update_directory` or something via command line separately -- defaulting to the well-known config file location in that case seems fine.

Reply:
👍 makes sense to me, thanks for the explanation!

Reply:
I will update the documentation to address this better as well!

Reply:
Done